nsxt/resource_nsxt_policy_gateway_policy_test.go

```diff
@@ -384,6 +384,204 @@ func TestAccResourceNsxtGlobalPolicyGatewayPolicy_withDomain(t *testing.T) {
 	})
 }
 
+func TestAccResourceNsxtPolicyGatewayPolicy_withIPCidrRangeDest(t *testing.T) {
+	name := getAccTestResourceName()
+	testResourceName := "nsxt_policy_gateway_policy.test"
+	defaultDirection := "IN_OUT"
+	defaultProtocol := "IPV4_IPV6"
+	policyIP := "10.10.20.5"
+	policyCidr := "10.10.20.0/22"
+	policyRange := "10.10.20.6-10.10.20.7"
+	updatedPolicyIP := "10.10.40.5"
+	updatedPolicyCidr := "10.10.40.0/22"
+	updatedPolicyRange := "10.10.40.6-10.10.40.7"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		CheckDestroy: func(state *terraform.State) error {
+			return testAccNsxtPolicyGatewayPolicyCheckDestroy(state, name, defaultDomain)
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: testAccNsxtPolicyGatewayPolicyWithIPCidrRangeDestCreate(name, policyIP, policyCidr, policyRange),
```

Comment:
I would suggest to use a single function to test both Create and Update use cases (just pass different values in params), and also I think one test should be enough to cover both source and dest use cases. While our goal is to have good test coverage, we also try to keep CI running time reasonable :)

Reply:
I went ahead and collapsed the tests into a single test for both security policy and gateway policy.  I also reduced the terraform generating functions down to single functions as you requested.